examples/launch_templates_with_managed_node_groups/templates/userdata.sh.tpl

```diff
@@ -0,0 +1,12 @@
+MIME-Version: 1.0
+Content-Type: multipart/mixed; boundary="//"
+
+--//
+Content-Type: text/x-shellscript; charset="us-ascii"
+#!/bin/bash 
+set -xe
+
+# Bootstrap and join the cluster
+/etc/eks/bootstrap.sh --b64-cluster-ca '${cluster_auth_base64}' --apiserver-endpoint '${endpoint}' ${bootstrap_extra_args} --kubelet-extra-args "${kubelet_extra_args}" '${cluster_name}'
+
+--//--
```

Comment:
I don't test it, can't we use directly the template from the module output `module.eks.workers_user_data` with the cloudinit provider ?

```hcl
data "cloudinit_config" "userdata" {
  base64_encode = true

  part {
    content_type = "text/x-shellscript"
    content      = module.eks.workers_user_data[0]
    filename     = "userdata.sh"
  }
}
```

Reply:
I have tried use this config to test but i am getting the following error
```

variable "eks_user_data" {
value = module.eks.workers_user_data[0]
}

data "cloudinit_config" "userdata" {
  base64_encode = true
  part {
    content_type = "text/x-shellscript"
    content      = var.eks_user_data
    filename     = "userdata.sh"
  }
}
```
Error in Terraform Plan

Error: Invalid index

 ```
 on source/_modules.tf line 246, in module "launch-templates":
 246:   eks_user_data = module.eks.workers_user_data[0]
    |----------------
    | module.eks.workers_user_data is empty tuple

The given key does not identify an element in this collection value.
```

Reply:
Just realized that templates are not generated for managed node group. So forget my comment. I'll open another PR to add that feature.